internal/store/device.go

```diff
@@ -27,7 +28,8 @@ type Device interface {
 	DeleteAll(ctx context.Context, orgId uuid.UUID, callback DeviceStoreAllDeletedCallback) error
 	Delete(ctx context.Context, orgId uuid.UUID, name string, callback DeviceStoreCallback) error
 	UpdateAnnotations(ctx context.Context, orgId uuid.UUID, name string, annotations map[string]string, deleteKeys []string) error
-	GetRendered(ctx context.Context, orgId uuid.UUID, name string, knownOwner, knownTemplateVersion *string) (*api.RenderedDeviceSpec, error)
+	UpdateRendered(ctx context.Context, orgId uuid.UUID, name string, rendered string) error
+	GetRendered(ctx context.Context, orgId uuid.UUID, name string, knownRenderedVersion *string) (*api.RenderedDeviceSpec, error)
```

Comment:
Optional

```suggestion
GetRendered(ctx context.Context, orgId uuid.UUID, name string, knownVersion *string) (*api.RenderedDeviceSpec, error)
```

Reply:
What's your reasoning?  It's not clear to me why this is better

Reply:
From an API perspective, you are creating a getter for `RenderedDeviceSpec`. As a parameter, you are asking for `knownRenderedVersion`. Rendered and known are already inferred. It is known if it is passed as a param?

I would argue that `RenderedDeviceSpec` should have a version instead of `RenderedVersion` because Rendered studders.  You have a `Version` of a `DeviceSpec` which is `Renderd`. Vs a `RenderedVersion` of a `RenderedDeviceSpec` as the version itself is not `Renderd`.

```suggestion
        GetRenderedSpec(ctx context.Context, orgId uuid.UUID, name string, version *string) (*api.RenderedDeviceSpec, error)
```

Reply:
I think @avishay is just trying to align with the RFE we approved. https://github.com/flightctl/flightctl/pull/247

I agree it stutters, and I am ok to change it, but I would do it as a follow up on the existing RFE+code to keep them aligned.

And by doing a follow up we also let avishay continue with the stream of changes

Reply:
OK can we create an issue to track?